File: 1400-construct-k-palindrome-strings/1400-construct-k-palindrome-strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if len(s) < k : return False
        freq = {}
        odd = 0
        
        for c in s:
            if c not in freq.keys():
                freq[c] = 1
            else: freq[c] += 1
                
        for _, item in freq.items():
            if item % 2: odd += 1
        
        return odd <= k

File: 1400-construct-k-palindrome-strings/1400-construct-k-palindrome-strings.c
bool canConstruct(char* s, int k) {
    if (strlen(s) < k) return 0;
    int cnt[26] = {0,};
    int odd = 0;

    for (int i = 0; i < strlen(s); i++) {
        cnt[s[i] - 'a']++;
    }
    
    for (int i = 0; i < 26; i++) {
        if (cnt[i] % 2) odd++;
    }
    
    return odd <= k;
}